wordpress-mobile/WordPress-Android
# My Site Dashboard - Phase 1 - Remove `MySiteFragment`

Fixes #15252

This PR removes the deprecated `MySiteFragment`, and also the feature flag `MySiteImprovementsFeatureConfig` used to guard the `ImprovedMySiteFragment` as we now always show `ImprovedMySiteFragment`.

Commits can be grouped as follows:
- Replace `MySiteFragment` with `ImprovedMySiteFragment` on bottom nav - ebf342b
- Remove references to`MySiteFragment` in `WPMainActivity` - e43f3e1 , 64a195c, f0772fc, 0cf03ec, 671a40c
- Delete `MySiteFragment` with its layout - 996d9cb, 669f8cf, 009c362, 64eaeeb
- Remove `MySiteImprovementsFeatureConfig.isEnabled()` guard - d5562ff, d407855, 5d3ee23, 4a0c025, 1c30cf2, 1418577, cf4ae26
- Remove duplicate tracking on complete quick start task/ tasks - 1418577 (See https://github.com/wordpress-mobile/WordPress-Android/issues/15250#issuecomment-907189777)

To test:
Smoke test "My Site" screen which now points to `ImprovedMySiteFragment`.

Also,
1. Notice that `App Settings` -> `Debug settings` -> `my_site_improvements_feature_flag` is not present anymore.
2. Toggle `QuickStartDynamicCardsFeatureConfig` for a site with "Quick Start" in progress and restart the app. Notice that the quick start section is appropriately shown based on the toggle state (with and without dynamic cards).

## Regression Notes
1. Potential unintended areas of impact
`MySiteFragment` was referenced from `WPMainActivity`, `Publicise`, `Reader` for showing quick start snackbars. It might be a good idea to test the quick start navigation with `ImprovedMySiteFragment` but there's a separate task to check `Quick Start - Guided Navigation` and we can add our findings to that task.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:
